### PR TITLE
[realsense2] Enable OpenNI2 driver option

### PR DESCRIPTION
**Pull request by UnaNancyOwen**
* Update realsense2 port to [librealsense v2.18.0](https://github.com/IntelRealSense/librealsense/releases/tag/v2.18.0).
* Add OpenNI2 driver option.

### CODE REVIEW

**Comment by UnaNancyOwen** [comment]
@vicroms Why did you do this operation?
What should I do something next?

**Comment by vicroms** [comment]
@UnaNancyOwen 

I had a local branch of this PR for testing that became outdated, I had to merge it with master in order to get regression results from the CI system. 

Please feel free to revert my commits if they had an adverse effect.

Also, the CI build reported regressions on x86, x64, and x64-static that I haven't been able to reproduce locally, which is strange... I'll queue a fresh build and wait for results.

**Comment by UnaNancyOwen** [comment]
@vicroms I fixed some problem.

**Comment by vicroms** [comment]
Hi @UnaNancyOwen 

Sorry for the long delay. I tried installing `realsense2[core,tools,openni2]:x64-windows` locally, but my build fails with the following log output:

It looks like the build is not finding the installed OpenNI2 headers.

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\CL.exe /c
    /I"D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\realsense2" 
    /I"D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\src" 
    /I"D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\include" 
    /I"D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\third-party\glfw\include" 
    /Z7 /W3 /WX- /diagnostics:classic /MP /O2 /Oi /D WIN32 /D _WINDOWS /D NDEBUG /D _UNICODE 
    /D NOMINMAX /D ELPP_THREAD_SAFE /D ELPP_NO_DEFAULT_LOG_FILE /D BUILD_EASYLOGGINGPP 
    /D HWM_OVER_XU /D ENFORCE_METADATA /D RS2_USE_WMF_BACKEND /D UNICODE /D __SSSE3__ 
    /D _CRT_SECURE_NO_WARNINGS /D "CMAKE_INTDIR=\"Release\"" /D _UNICODE /D UNICODE /Gm- 
    /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"rs-color.dir\Release\\" 
    /Fd"rs-color.dir\Release\vc141.pdb" /Gd /TC /FC /errorReport:queue  /utf-8 
    "D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\examples\C\color\rs-color.c"
.
.
.
30> D:\vcpkg\buildtrees\realsense2\src\v2.18.0-5ace468cc5\wrappers\openni2\src\rs2base.h(3): 
    fatal error C1083:
        Cannot open include file: 'Driver/OniDriverAPI.h': 
        No such file or directory (compiling source file 
            D:\vcpkg\buildtrees\realsense2\src\v2.18.05ace468cc5\wrappers\openni2\src\Rs2Base.cpp)
```

**Comment by UnaNancyOwen** [comment]
@vicroms Thanks, I fixed it. Please re-confirm.

**Comment by vicroms** [comment]
Looks like that fixed the build with the OpenNI feature, however we still have problems in the CI machine.

Specifically, it fails in the configure step when trying to fetch `libusb` from `git`.
Can you make it so that it depends on the `libusb` installed by `vcpkg`?

```
-- Use external libusb
CMake Error at D:/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/share/cmake-3.12/Modules/ExternalProject.cmake:2324 (message):
  error: could not find git for clone of libusb
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMake/external_libusb.cmake:4 (ExternalProject_Add)
  CMake/windows_config.cmake:41 (include)
  CMakeLists.txt:17 (os_set_flags)
```

**Comment by UnaNancyOwen** [comment]
libusb port can not be installed on my environments.
Were you able to install it? <code>vcpkg install libusb:x64-windows</code>

**Comment by vicroms** [comment]
I'm able to install `libusb:x64-windows`.

I'm using the latest portfile version from master `1.0.22-1`.
Can you see why it fails for you in the output logs?

```
PS D:\github\vcpkg> ./vcpkg install libusb:x64-windows --no-binarycaching
The following packages will be built and installed:
    libusb[core]:x64-windows
Starting package 1/1: libusb:x64-windows
Building package libusb[core]:x64-windows...
-- Downloading https://github.com/libusb/libusb/archive/v1.0.22.tar.gz...
-- Extracting source D:/github/vcpkg/downloads/libusb-libusb-v1.0.22.tar.gz
-- Using source at D:/github/vcpkg/buildtrees/libusb/src/v1.0.22-f45a56be59
-- Building msvc/libusb_dll_2017.vcxproj for Release
-- Building msvc/libusb_dll_2017.vcxproj for Debug
-- Installing: D:/github/vcpkg/packages/libusb_x64-windows/share/libusb/copyright
-- Installing: D:/github/vcpkg/packages/libusb_x64-windows/include/libusb-1.0/libusb.h
-- Performing post-build validation
-- Performing post-build validation done
Building package libusb[core]:x64-windows... done
Installing package libusb[core]:x64-windows...
Installing package libusb[core]:x64-windows... done
```

**Comment by UnaNancyOwen** [comment]
It is this error https://github.com/libusb/libusb/issues/207. 
It seems to be occure problem on multi-byte character locale such as Japanese.
I think that it should make the same fix as [this patch](https://github.com/Microsoft/vcpkg/pull/5275/files#diff-4367c06610efb0af0d42bb12e3ef16b9) to libusb port.
```
c:\vcpkg\buildtrees\libusb\x64-windows-rel\v1.0.22-f45a56be59\libusb\strerror.c(111): error C2001: newline in constant [C:\vcpkg\buildtrees\libusb\x64-windows-rel\v1.0.22-f45a56be59\msvc\libusb_dll_2017.vcxproj]
c:\vcpkg\buildtrees\libusb\x64-windows-rel\v1.0.22-f45a56be59\libusb\strerror.c(117): error C2001: newline in constant [C:\vcpkg\buildtrees\libusb\x64-windows-rel\v1.0.22-f45a56be59\msvc\libusb_dll_2017.vcxproj]
```

**Comment by UnaNancyOwen** [comment]
> Can you make it so that it depends on the libusb installed by vcpkg?

@vicroms I tried it, but I gave up because it was difficult for me at least. :'(

**Comment by vicroms** [comment]
Update to 2.19 is being handled in PR #5777 


@UnaNancyOwen 
Can the scope of this PR be reduced to adding NI driver support?
I seem to recall that working OK at least locally.

**Comment by UnaNancyOwen** [comment]
@vicroms No, OpenNI2 driver is included since RealSense SDK v2.16.2 (or later). However, This port is RealSense SDK v2.16.1 in the current vcpkg (master/HEAD).
Therefore, It is necessary to update RealSense SDK to v2.16.2 (or later) to implement OpenNI2 driver option.  (I suggest to implement this option after merged #5777. )

**Comment by vicroms** [comment]
@UnaNancyOwen 

Thanks for the PR!

File: ports/realsense2/portfile.cmake [hunk 1]
```diff
@@ -63,5 +70,11 @@ if(BUILD_EXAMPLES)
     endif()
 endif()
 
+
+if(BUILD_OPENNI2_BINDINGS)
+  file(COPY ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/_out/rs2driver*
```
Comment (Voskrese) [inline ports/realsense2/portfile.cmake]: error

> CMake Error at ports/realsense2/portfile.cmake:75 (file):
>   file COPY cannot find
>   "E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver*".
> Call Stack (most recent call first):
>   scripts/ports.cmake:74 (include)

`ls E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver*`

> E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver.dll
> E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver.dll.manifest
> E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver.exp
> E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver.lib
> E:/tools/vcpkg/buildtrees/realsense2/x64-windows-rel/_out/rs2driver.pdb


copy **tools** /openni2 ? *.lib? why?

**Pull request merged**